### PR TITLE
Marine animation integration

### PR DESCRIPTION
**Pull request by brodymccrone**
-Changed marine image update to use delta velocities instead of keyboard input.
-Changed PlayerData struct to contain delta velocities.
-Added support for deltas in PlayerData in server's getPlayers
-Add support for deltas in client's updateMarine
-Moved image update from GameStateMatch::handle to updateMarines.

### CODE REVIEW

**Review by eyyu — APPROVED** [review]

File: src/game/GameManager.cpp [hunk 1]
```diff
@@ -165,14 +171,14 @@ bool GameManager::createMarine(const float x, const float y) {
 }
 
 void GameManager::createMarine(const int32_t id) {
-    SDL_Rect temp = {INITVAL, INITVAL, MARINE_WIDTH, MARINE_HEIGHT};
+    SDL_Rect temp = {INITVAL, INITVAL, DEFAULT_SIZE, DEFAULT_SIZE};
```
Comment (JohnAgapeyev) [inline src/game/GameManager.cpp]: Why was this changed back?
All other marines are created with the MARINE globals

File: src/game/GameManager.cpp [hunk 2]
```diff
@@ -165,14 +171,14 @@ bool GameManager::createMarine(const float x, const float y) {
...
 
     SDL_Rect marineRect = temp;
     SDL_Rect moveRect = temp;
     SDL_Rect projRect = temp;
     SDL_Rect damRect = temp;
 
-    marineManager.emplace(id, Marine(id, marineRect, moveRect, projRect, damRect));
+    marineManager.insert({id, Marine(id, marineRect, moveRect, projRect, damRect)});
```
Comment (JohnAgapeyev) [inline src/game/GameManager.cpp]: No, just no
Undo this change

File: src/game/GameManager.cpp [hunk 3]
```diff
@@ -563,6 +569,8 @@ void GameManager::updateCollider() {
         }
     }
 
+
+
```
Comment (JohnAgapeyev) [inline src/game/GameManager.cpp:706]: Not sure we need all this whitespace

**Review by JohnAgapeyev — CHANGES_REQUESTED** [review]
Couple things changed that probably shouldn't have

**Review by JohnAgapeyev — APPROVED** [review]
